### PR TITLE
Example folders

### PR DESCRIPTION
**Pull request by Crissman**
## Motivation
Responding to issue #1238.

## Description of the changes
Create folders for each platform that has more than one example to reduce the number of files in the examples folder.

Note, on Merge, optuna.org and Medium posts that reference these examples should be changed.

More updating of URLs will be necessary with this PR. Posting first as WIP to confirm ok to make the directory changes.

### CODE REVIEW

**Comment by HideakiImamura** [comment]
@toshihikoyanase Could you review this PR if you have time?

**Comment by toshihikoyanase** [comment]
Sure!

**Comment by Crissman** [comment]
@toshihikoyanase @HideakiImamura 🙇‍♂️

File: examples/README.md [hunk 1]
```diff
@@ -9,25 +9,25 @@ This page contains a list of example codes written with Optuna.
...
-* [XGBoost](./xgboost_simple.py)
-* [Tensorflow](./tensorflow_estimator_simple.py)
-* [Tensorflow (eager)](./tensorflow_eager_simple.py)
+* [PyTorch](./pytorch/pytorch_simple.py)
+* [PyTorch Ignite](./pytorch/pytorch_ignite_simple.py)
+* [PyTorch Lightning](./pytorch/pytorch_lightning_simple.py)
+* [XGBoost](./xgboost/xgboost_simple.py)
+* [Tensorflow](./tensorflow/tensorflow_estimator_simple.py)
+* [Tensorflow (eager)](./tensorflow/tensorflow_eager_simple.py)
 * [Keras](./keras_simple.py)
```
Comment (HideakiImamura) [inline examples/README.md:28]:
```suggestion
* [Keras](./keras/keras_simple.py)
```

File: examples/pytorch/pytorch_lightning_simple.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
+"""
```
Comment (HideakiImamura) [inline examples/pytorch/pytorch_lightning_simple.py:1]: Should we remove `examples/pytorch_lightning_simple.py`?

File: examples/README.md [hunk 3]
```diff
@@ -9,25 +9,25 @@ This page contains a list of example codes written with Optuna.
 
 ### Examples with ML Libraries
 
-* [Scikit-learn](./sklearn_simple.py)
+* [Scikit-learn](./sklearn/sklearn_simple.py)
```
Comment (HideakiImamura) [inline examples/README.md:12]: The `OptunaSearchCV` used in `examples/optuna_search_cv_simple.py` is the integration module with `scikit-learn`. How about placing it into `examples/sklearn` and adding here?

**Review by HideakiImamura — COMMENTED** [review]
Thanks for the update. I have several comments. Could you take a look?

I've noticed the following points should be changed.
- `optuna/integration/_lightgbm_tuner/__init__.py:L19`
- `optuna/integration/tensorflow.py:L18`

**Comment by Crissman** [comment]
Thanks for the catches, @HideakiImamura . Implemented! 🙇‍♂️

File: optuna/integration/_lightgbm_tuner/__init__.py [hunk 4]
```diff
@@ -16,8 +16,9 @@ def train(*args: Any, **kwargs: Any) -> Any:
 
     It tunes important hyperparameters (e.g., ``min_child_samples`` and ``feature_fraction``) in a
     stepwise manner. It is a drop-in replacement for `lightgbm.train()`_. See
-    `a simple example of LightGBM Tuner <https://github.com/optuna/optuna/blob/master/examples/lig
-    htgbm_tuner_simple.py>`_ which optimizes the validation log loss of cancer detection.
+    `a simple example of LightGBM Tuner <https://github.com/optuna/optuna/blob/master/examples/
+    ligthgbm/lig htgbm_tuner_simple.py>`_ which optimizes the validation log loss of cancer
```
Comment (toshihikoyanase) [inline optuna/integration/_lightgbm_tuner/__init__.py:20]: Could you fix the path?
```suggestion
    lightgbm/lightgbm_tuner_simple.py>`_ which optimizes the validation log loss of cancer
```

File: optuna/integration/tensorflow.py [hunk 5]
```diff
@@ -15,7 +15,7 @@ class TensorFlowPruningHook(SessionRunHook):
     """TensorFlow SessionRunHook to prune unpromising trials.
 
     See `the example <https://github.com/optuna/optuna/blob/master/examples/
-    pruning/tensorflow_estimator_integration.py>`_
+    tensorflow/tensorflow/tensorflow_estimator_integration.py>`_
```
Comment (toshihikoyanase) [inline optuna/integration/tensorflow.py:18]: ditto.
```suggestion
    tensorflow/tensorflow_estimator_integration.py>`_
```

File: optuna/pruners/_successive_halving.py [hunk 6]
```diff
@@ -19,10 +19,10 @@ class SuccessiveHalvingPruner(BasePruner):
 
     Note that, this class does not take care of the parameter for the maximum
     resource, referred to as :math:`R` in the paper. The maximum resource allocated to a trial is
-    typically limited inside the objective function (e.g., ``step`` number in `simple.py
-    <https://github.com/optuna/optuna/tree/c5777b3e/examples/pruning/simple.py#L31>`_,
+    typically limited inside the objective function (e.g., ``step`` number in `simple_pruning.py
+    <https://github.com/optuna/optuna/tree/c5777b3e/examples/simple_pruning.py#L31>`_,
```
Comment (toshihikoyanase) [inline optuna/pruners/_successive_halving.py:23]: This URL specifies the commit id, so please leave it as it is. Or, we can replace it with the master branch as follows:
```suggestion
    <https://github.com/optuna/optuna/blob/master/examples/simple_pruning.py#L31>`_,
```

File: optuna/pruners/_successive_halving.py [hunk 7]
```diff
@@ -19,10 +19,10 @@ class SuccessiveHalvingPruner(BasePruner):
...
     ``EPOCH`` number in `chainer_integration.py
-    <https://github.com/optuna/optuna/tree/c5777b3e/examples/pruning/chainer_integration.py#L65>`_).
+    <https://github.com/optuna/optuna/tree/c5777b3e/examples/chainer/chainer_integration.py#L65>`_).
```
Comment (toshihikoyanase) [inline optuna/pruners/_successive_halving.py:25]: ditto.
```suggestion
    <https://github.com/optuna/optuna/blob/master/examples/chainer/chainer_integration.py#L65>`_).
```

**Review by toshihikoyanase — CHANGES_REQUESTED** [review]
Thank you for your PR. I found some wrong links. Please take a look.

File: examples/README.md (hunk 3 above)
Comment (HideakiImamura) [reply examples/README.md:12]: The name of class is `OptunaSearchCV`, so how about naming the `sklearn_optuna_search_cv_simple.py`? In addition, how about adding the file to this line.

**Comment by Crissman** [comment]
I'm impressed with both of your meticulous checks. 🙇  Thanks!

**Comment by HideakiImamura** [comment]
@Crissman Thanks for the update! Almost LGTM, but you may miss this https://github.com/optuna/optuna/pull/2302#discussion_r573119688.

**Review by HideakiImamura — APPROVED** [review]
LGTM.

File: examples/pytorch/pytorch_lightning_simple.py [hunk 8]
```diff
@@ -17,6 +17,7 @@
 
 from packaging import version
 import pytorch_lightning as pl
+from pytorch_lightning import Callback
```
Comment (toshihikoyanase) [inline examples/pytorch/pytorch_lightning_simple.py:20]: I think you revert the changes in #2294. Could you remove the changes in this file?

**Review by toshihikoyanase — APPROVED** [review]
Thank you for your swift action. LGTM!

**Pull request merged**